OfflineUX.md

```diff
@@ -0,0 +1,138 @@
+</div>
+
+The example we just looked at is nice and simple, but some actions should use this apporach (example: requesting money from another user). For these types of actions, we can’t simply proceed as if the request already finished. Here are some reasons why:
+
+1. In some instances, we simply wouldn't know _how_ to proceed because of a lack of information (often the server returns data that we wouldn't be able to guess the content of). 
+
+2. In other cases, we may be able to guess what a successful request would look like, but we don't want to misguide the user into believing their action was completed. For example, we don't want the user to believe that a financial transaction has been made when it actually hasn't. 
+
+To handle problems like this, we have developed offline patterns and guidance on when to use them. These are called the Offline UX Patterns, which are explained below.
+
+### Descriptions of the UX Patterns
+
+# None - No Offline Behavior
+
+ There’s no specific UI for this case. The feature either looks totally normal and works as expected (because it doesn’t need the server to function) or the feature looks like it did whenever connection was lost. 
+
+Used when…
+ - there is no interaction with the server in any way
+ - or data is READ from the server and does not need to show up-to-date data. The user will see stale data until the new data is put into Onyx and then the view updates to show the new data. 
+
+# A - Optimistic Without Feedback Pattern
+
+This is the pattern where we queue the request to be sent when the user is online and we continue as if the request succeeded. 
+
+Used when…
+ - the user should be given instant feedback and
+ - there is no error handling in the server and
+ - the user does not need to know when the change is done on the server in the background
+
+How to implement: Use API.write() to implement this pattern. 
+
+# B - Optimistic WITH Feedback Pattern
+This pattern queues the API request, but also makes sure that the user is aware that the request hasn’t been sent yet **when the user is offline**. When the user is online, the feature should just look like it succeeds immediately (we dont want the offline UI to flicker on and off when the user is online).
+
+Used when…
+ - the user needs feedback that data is being sent to the server
+This is a minority use case at the moment, but INCREDIBLY HELPFUL for the user, so proceed with cautious optimism.
+
+How to implement: Use API.write() to implement this pattern. 
+
+# C - Blocking Form UI Pattern
+This pattern blocks the user from interacting with an entire page.
+
+Used when…
+ - a form is used to make a WRITE request to the server and 
+ - server has to do some validation of the parameters that can’t be done in the client or
+ - server response will be unknown so it cannot be done optimistically
+ - If the request is moving money 
```

Comment:
I'm not sure what you're saying here. Can you clarify what your suggestion is?

Reply:
Yeah, I'm suggesting that we remove this list item, because `If the request is moving money` does not necessarily mean that we'll be using the blocking form UX pattern.

cc @iwiznia because looking at previous threads it looks like you suggested we add this line